src/box/xrow_update_array.c

```diff
@@ -393,8 +426,11 @@ xrow_update_op_do_array_delete(struct xrow_update_op *op,
 
 	struct xrow_update_rope *rope = field->array.rope;
 	uint32_t size = xrow_update_rope_size(rope);
-	if (xrow_update_op_adjust_field_no(op, size) != 0)
+	if (xrow_update_op_adjust_field_no(op, size) != 0) {
```

Comment:
This is a separate behaviour change. Better do it in another commit with its own tests.